services/app-users/src/user/user.controller.ts

```diff
@@ -50,4 +50,16 @@ export class UserController {
     console.log(`${this._name} handleBulkDelete Payload`, payload);
     return this.userService.handleBulkDelete(payload);
   }
+  @MessagePattern(USER_EVENTS.login)
+  async getMessage(payload: TMessagePayload) {
+    console.log(`${this._name} login payload`, payload);
+    return this.userService.loginUser(payload);
+  }
+
+  //In case there is a need to end-to-end encrypted message
```

Comment:
End to end encryption can be handled in the guards. The applicationRoles.guard.ts will need to be moved to the shared-modukes lib.